### PR TITLE
MCR-85 Refactor menu.py 

### PR DESCRIPTION
**Pull request by jasonlwj**
My work is far from done but I figured current changes can be merged in for now.

Part 2 coming soon (soon)

### CODE REVIEW

**Comment by ChrisNeedham24** [comment]
This will partially address #85.

**Comment by jasonlwj** [comment]
I bet the linting is gonna fail my PR

File: menu.py [hunk 1]
```diff
@@ -242,15 +243,11 @@ def draw(self):
...
-                            pyxel.text(25, 75, "Like any strong leader, you want the", pyxel.COLOR_WHITE)
-                            pyxel.text(25, 81, "best for your people. However,", pyxel.COLOR_WHITE)
-                            pyxel.text(25, 87, "constant attacks by filthy Heathens", pyxel.COLOR_WHITE)
-                            pyxel.text(25, 93, "and enemy troops are enough to wear", pyxel.COLOR_WHITE)
-                            pyxel.text(25, 99, "any great leader down. It is time to", pyxel.COLOR_WHITE)
-                            pyxel.text(25, 105, "put an end to this, and become the", pyxel.COLOR_WHITE)
-                            pyxel.text(25, 111, "one true empire. Other empires will", pyxel.COLOR_WHITE)
-                            pyxel.text(25, 117, "wither at your blade, and they will", pyxel.COLOR_WHITE)
-                            pyxel.text(25, 123, "be all the more thankful for it.", pyxel.COLOR_WHITE)
+                            self.draw_paragraph(25, 75, """Like any strong leader, you want the best for your people. However,
```
Comment (ChrisNeedham24) [inline menu.py:246]: Would advise running pylint, this line and many others go over the 120 char limit :(

File: menu.py [hunk 2]
```diff
@@ -1009,3 +797,85 @@ def get_game_config(self) -> GameConfig:
         """
         return GameConfig(self.player_count, self.faction_colours[self.faction_idx][0], self.biome_clustering_enabled,
                           self.fog_of_war_enabled, self.climatic_effects_enabled)
+
+    def next_menu_option(self, current_option: typing.Type[Enum], wrap_around: bool = False) -> None:
```
Comment (ChrisNeedham24) [inline menu.py:801]: Getting some warnings about this Enum type hint - might be easier to change it from `typing.Type[Enum]` to `SetupOption | WikiOption | MainMenuOption | VictoryType`?

File: menu.py [hunk 3]
```diff
@@ -357,7 +331,7 @@ def draw(self):
                     pyxel.rect(21, 11, 158, 182, pyxel.COLOR_BLACK)
                     pyxel.text(85, 15, "Factions", pyxel.COLOR_WHITE)
                     pyxel.text(25, 30, str(self.faction_colours[self.faction_wiki_idx][0].value),
-                               self.faction_colours[self.faction_wiki_idx][1])
+                            self.faction_colours[self.faction_wiki_idx][1])
```
Comment (ChrisNeedham24) [inline menu.py:334]: While you're fixing the lint, this indentation change should be reverted.

File: menu.py [hunk 4]
```diff
@@ -1009,3 +797,85 @@ def get_game_config(self) -> GameConfig:
...
+            case SetupOption():
+                self.setup_option = target_option
+            case WikiOption():
+                self.wiki_option = target_option
+            case MainMenuOption():
+                self.main_menu_option = target_option
+            case VictoryType():
+                self.victory_type = target_option
+
+    @staticmethod
```
Comment (ChrisNeedham24) [inline menu.py:855]: Having this as a staticmethod is cool for now, but I suspect we'll eventually break this (and other helper methods, maybe the above ones) out into a util file without a class.

**Review by ChrisNeedham24 — CHANGES_REQUESTED** [review]
Functionality all works great, love the helper methods, but the above threads will need to be addressed.

**Comment by jasonlwj** [comment]
- [ ] [841bd66](https://github.com/ChrisNeedham24/microcosm/pull/91/commits/841bd66dbd0cc1c655d54ad409b59603d697ddd2)

File: menu.py (hunk 3 above)
Comment (jasonlwj) [reply menu.py]: [841bd66](https://github.com/ChrisNeedham24/microcosm/pull/91/commits/841bd66dbd0cc1c655d54ad409b59603d697ddd2)

File: menu.py (hunk 2 above)
Comment (jasonlwj) [reply menu.py]: [841bd66](https://github.com/ChrisNeedham24/microcosm/pull/91/commits/841bd66dbd0cc1c655d54ad409b59603d697ddd2)

File: menu.py (hunk 1 above)
Comment (jasonlwj) [reply menu.py]: [841bd66](https://github.com/ChrisNeedham24/microcosm/pull/91/commits/841bd66dbd0cc1c655d54ad409b59603d697ddd2)

File: menu.py [hunk 5]
```diff
@@ -331,7 +334,7 @@ def draw(self):
                     pyxel.rect(21, 11, 158, 182, pyxel.COLOR_BLACK)
                     pyxel.text(85, 15, "Factions", pyxel.COLOR_WHITE)
                     pyxel.text(25, 30, str(self.faction_colours[self.faction_wiki_idx][0].value),
-                            self.faction_colours[self.faction_wiki_idx][1])
+                        self.faction_colours[self.faction_wiki_idx][1])
```
Comment (ChrisNeedham24) [inline menu.py:337]: I think this has actually made it worse but it's fine lol

Comment (jasonlwj) [reply menu.py:337]: oh no i forgor 💀

Comment (ChrisNeedham24) [reply menu.py]: [c8f4e20](https://github.com/ChrisNeedham24/microcosm/pull/91/commits/c8f4e200da76f53b0aea67e242812e7d0ff8f24d)

File: menu.py (hunk 3 above)
Comment (ChrisNeedham24) [reply menu.py]: [c8f4e20](https://github.com/ChrisNeedham24/microcosm/pull/91/commits/c8f4e200da76f53b0aea67e242812e7d0ff8f24d)

**Review by ChrisNeedham24 — APPROVED** [review]
Tasks addressed nicely, great refactor 🙌

**Pull request merged**